### PR TITLE
【KernelGen】Add top_k_per_row_decode operator for vLLM speculative decoding

### PR DESCRIPTION
**Pull request by Schopenhauer-loves-Hegel**
Implement histogram-based radix select O(n) algorithm for efficient top-k selection:
- Uses 256-bin histogram with 4-round refinement (8 bits per round)
- Handles boundary cases where valid_count < K
- Fixed block sizes to avoid Triton JIT non-determinism

### PR Category
Operator

### Type of Change
New Feature

### Description
Add `top_k_per_row_decode` operator for vLLM's speculative decoding phase. This operator selects top-k indices from each row of logits tensor using histogram-based radix select algorithm with O(n) complexity.

Key features:
- Float32 to uint32 order-preserving conversion for histogram binning
- 4-round refinement using 8 bits per round (32 bits total)
- Compile-time constant loop bounds to avoid Triton JIT non-determinism
- Proper handling of boundary cases (valid_count < K)

### Issue
N/A

### Progress

- [ ] Change is properly reviewed (1 reviewer required, 2 recommended).
- [ ] Change is responded to an issue.
- [x] Change is fully covered by a UT.

### Performance
benchmark/test_vllm_perf.py::test_top_k_per_row_decode_benchmark
Operator: top_k_per_row_decode  Performance Test (dtype=small_rows, mode=kernel,level=comprehensive)
Status       Torch Latency (ms)    Gems Latency (ms)         Gems Speedup          Size Detail
-----------------------------------------------------------------------------------------------
SUCCESS               0.023840            0.073312               0.325          [torch.Size([32, 20000]), 1, torch.Size([32]), torch.Size([32, 10]), 32, 20000, 1, 10]
SUCCESS               0.071520            0.393888               0.182          [torch.Size([32, 128256]), 1, torch.Size([32]), torch.Size([32, 10]), 32, 128256, 1, 10]

Operator: top_k_per_row_decode  Performance Test (dtype=medium_rows, mode=kernel,level=comprehensive)
Status       Torch Latency (ms)    Gems Latency (ms)         Gems Speedup          Size Detail
-----------------------------------------------------------------------------------------------
SUCCESS               0.025472            0.082208               0.310          [torch.Size([128, 20000]), 1, torch.Size([128]), torch.Size([128, 10]), 128, 20000, 1, 10]
SUCCESS               0.099232            0.479104               0.207          [torch.Size([128, 128256]), 1, torch.Size([128]), torch.Size([128, 10]), 128, 128256, 1, 10]

Operator: top_k_per_row_decode  Performance Test (dtype=large_rows, mode=kernel,level=comprehensive)
Status       Torch Latency (ms)    Gems Latency (ms)         Gems Speedup          Size Detail
-----------------------------------------------------------------------------------------------
SUCCESS               0.114912            0.297376               0.386          [torch.Size([1024, 20000]), 1, torch.Size([1024]), torch.Size([1024, 10]), 1024, 20000, 1, 10]
SUCCESS               0.448560            1.784608               0.251          [torch.Size([1024, 128256]), 1, torch.Size([1024]), torch.Size([1024, 10]), 1024, 128256, 1, 10]
SUCCESS               0.209184            0.564560               0.371          [torch.Size([2048, 20000]), 1, torch.Size([2048]), torch.Size([2048, 10]), 2048, 20000, 1, 10]
SUCCESS               0.862688            3.403584               0.253          [torch.Size([2048, 128256]), 1, torch.Size([2048]), torch.Size([2048, 10]), 2048, 128256, 1, 10]
SUCCESS               0.395072            1.095808               0.361          [torch.Size([4096, 20000]), 1, torch.Size([4096]), torch.Size([4096, 10]), 4096, 20000, 1, 10]
SUCCESS               1.665696            6.525312               0.255          [torch.Size([4096, 128256]), 1, torch.Size([4096]), torch.Size([4096, 10]), 4096, 128256, 1, 10]
SUCCESS               0.761056            2.172624               0.350          [torch.Size([8192, 20000]), 1, torch.Size([8192]), torch.Size([8192, 10]), 8192, 20000, 1, 10]
SUCCESS               3.235440           12.814656               0.252          [torch.Size([8192, 128256]), 1, torch.Size([8192]), torch.Size([8192, 10]), 8192, 128256, 1, 10]

### CODE REVIEW

**Comment by bin913** [comment]
@Schopenhauer-loves-Hegel Please Resolve conflicts

**Comment by bin913** [comment]
@Schopenhauer-loves-Hegel Is this PR still needed? Can it be closed?

**Comment by lukatao** [comment]
@bin913 Closing this PR — upstream master already has a superior implementation of `top_k_per_row_decode` with multi-tier dispatch (single/medium/large block) and much better performance. This histogram-based version is now superseded. Thanks for the follow-up!
